pints/_mcmc/_dual_averaging.py

```diff
@@ -0,0 +1,241 @@
+#
+# Dual Averaging step size and mass matrix adaption method for NUTS and HMC
+# samplers
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+
+
+class DualAveragingAdaption:
+    """
+    Implements a Dual Averaging scheme to adapt the step size ``epsilon``, as
+    per [1]_, and estimates the (fully dense) inverse mass matrix using the
+    sample covariance of the accepted parameter, as suggested in [2]_
+
+    The adaption is done using the same windowing method employed by STAN,
+    which is done over three or more windows:
+
+    - initial window: epsilon is adapted using dual averaging
+    - base window: epsilon continues to be adapted using dual averaging, this
```

Comment:
Not sure I follow the distinction between the base window and the initial window. Is the only difference that (at the end of the base window), the mass matrix is updated?

Reply:
Yes, you are correct. I've clarified this in the docstring.